person.rb

```diff
@@ -1,10 +1,12 @@
-class Person
-  attr_accessor :name, :age
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :age
   attr_reader :id
 
   def initialize(age, parent_permission: true, name: 'Unknown')
+    super(name)
     @id = rand(1..1000)
-    @name = name
     @age = age
     @parent_permission = parent_permission
   end
```

Comment:
- [ ] I was having difficulty testing your decoration. Kindly remove `name` from the super method and add the initial line you had before. You should have something like this

  ```
  def initialize(age, parent_permission: true, name: 'Unknown')
    super()
    @id = rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  ```